src/connectors/joox.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+
+
+const defaultAlbumData = {
+	path: undefined,
+	loaded: false,
+	name: undefined
+};
+
+const playerSelector = '#jooxPlayerBar';
+const artistSelector = '#lblSinger';
+const trackSelector = '#lblSongName';
+const playBtnSelector = '#btnPlay';
+
+let _albumData = $.extend({}, defaultAlbumData);
+
+const getTrackPath = () => $(trackSelector).attr('u');
+const getTrackUrl = () => `https://www.joox.com/mm/en/${getTrackPath()}`;
+const initiateAlbumDataCache = (trackPath) => $.extend({}, defaultAlbumData, { path: trackPath });
+const isAlbumDataInCache = (albumData, trackPath) => albumData.path === trackPath;
+const isAlbumDataLoaded = (albumData) => albumData.loaded;
+const loadAlbumData = (onLoadComplete) => {
+	$.get({
+		url: getTrackUrl(),
+		success: (data) => {
+			const itemIndex = 1; // it's wrongly tagged in the HTML of joox. We may need it changed when they fixed the bug.
+			const element = $(data).find('[itemtype="https://schema.org/MusicRecording"] [itemprop="name"]')[itemIndex];
+			onLoadComplete($(element).text());
+		},
+	});
+	return true;
+};
+
+const filter = new MetadataFilter({
+	track: zg2uni,
+	album: zg2uni
+});
+
+Connector.playerSelector = playerSelector;
+Connector.trackSelector = trackSelector;
+Connector.currentTimeSelector = '#lblCurrentTime';
+Connector.durationSelector = '#lblTotalTime';
+Connector.trackArtSelector = '#jooxAlbum';
+Connector.getUniqueID = getTrackUrl;
+
+Connector.getArtist = () => {
+	const _element = $(artistSelector);
+	return _element.attr('title') || _element.text();
+};
+
+Connector.isPlaying = () => $(playBtnSelector).hasClass('joox-pause');
+
+Connector.getAlbum = () => {
+	const trackPath = getTrackPath();
+
+	if (!isAlbumDataInCache(_albumData, trackPath)) {
+		_albumData = initiateAlbumDataCache(trackPath);
+	}
+
+	if (!isAlbumDataLoaded(_albumData)) {
+		_albumData.loaded = loadAlbumData((album_name) => _albumData.name = album_name);
+	}
+
+	return _albumData.name;
+};
+
+Connector.applyFilter(filter);
+
+// minified zg2uni of Rabbit converter
```

Comment:
To me this feels like it should be included as a 3rd party dependency and available not just here.